pkg/apis/core/v1/conversion.go

```diff
@@ -156,7 +157,8 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 	// Add field conversion funcs.
 	err = scheme.AddFieldLabelConversionFunc("v1", "Pod",
 		func(label, value string) (string, string, error) {
-			switch label {
+			path, _ := fieldpath.SplitMaybeSubscriptedPath(label)
```

Comment:
is there a reason to make this pod-specific, rather than supporting it for all objects in DefaultMetaV1FieldSelectorConversion?

Reply:
I'm not aware of this. The design doc that this PR implements is for pod only. Can you please give an example of other API objects that can use this feature?

Reply:
the place this is being added is in the path for field selector conversion...

this makes the server accept requests like `kubectl get pods "--field-selector=metadata.annotations['foo']=bar"` (though it doesn't actually satisfy those requests correctly)

since all objects have labels and annotations, I'd expect a feature allowing field selectors to use those fields to apply equally to all objects